File: packages/intune/android/src/main/java/io/capawesome/capacitorjs/plugins/intune/classes/options/IsFileEncryptedOptions.java
package io.capawesome.capacitorjs.plugins.intune.classes.options;

import androidx.annotation.NonNull;
import com.getcapacitor.PluginCall;
import io.capawesome.capacitorjs.plugins.intune.IntuneHelper;
import io.capawesome.capacitorjs.plugins.intune.classes.CustomExceptions;

public class IsFileEncryptedOptions {

    @NonNull
    private final String path;

    public IsFileEncryptedOptions(@NonNull PluginCall call) throws Exception {
        this.path = IsFileEncryptedOptions.getPathFromCall(call);
    }

    @NonNull
    public String getPath() {
        return path;
    }

    @NonNull
    private static String getPathFromCall(@NonNull PluginCall call) throws Exception {
        String path = call.getString("path");
        if (path == null) {
            throw CustomExceptions.PATH_MISSING;
        }
        return IntuneHelper.getFilePath(path);
    }
}

File: packages/intune/ios/Plugin/Classes/Options/IsFileEncryptedOptions.swift
import Foundation
import Capacitor

@objc public class IsFileEncryptedOptions: NSObject {
    let path: String

    init(_ call: CAPPluginCall) throws {
        self.path = try IsFileEncryptedOptions.getPathFromCall(call)
    }

    private static func getPathFromCall(_ call: CAPPluginCall) throws -> String {
        guard let path = call.getString("path") else {
            throw CustomError.pathMissing
        }
        return IntuneHelper.getFilePath(path)
    }
}
